wire-ios-sync-engine/Tests/Source/Use cases/FetchShareableConversationUseCaseTests.swift

```diff
@@ -0,0 +1,81 @@
+//
+// Wire
+// Copyright (C) 2024 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import XCTest
+
+@testable import WireDataModelSupport
+@testable import WireSyncEngine
+
+class FetchShareableConversationUseCaseTests: XCTestCase {
+
+    // MARK: - Properties
+
+    private var sut: FetchShareableConversationsUseCase!
+    private var coreDataStackHelper: CoreDataStackHelper!
+    private var coreDataStack: CoreDataStack!
+
+    // MARK: - Life cycle
+
+    override func setUp() async throws {
+        try await super.setUp()
+
+        coreDataStackHelper = CoreDataStackHelper()
+        coreDataStack = try await coreDataStackHelper.createStack()
+        sut = FetchShareableConversationsUseCase(contextProvider: coreDataStack)
+    }
+
+    override func tearDown() {
+        sut = nil
+        coreDataStack = nil
+        coreDataStackHelper = nil
+        super.tearDown()
+    }
+
+    // MARK: - Tests
+
+    func testInvoke() {
+        // Given
+        let selfUser = ZMUser.selfUser(in: coreDataStack.viewContext)
+
+        // create a shareable conversation
+        let shareable = createConversation()
+        shareable.addParticipantAndUpdateConversationState(user: selfUser)
+
+        // create a non-shareable conversation
+        let nonShareable = createConversation()
```

Comment:
Do you get a warning here for an unused type?

Reply:
Updated https://github.com/wireapp/wire-ios/pull/1658/commits/8c85d9ddbe3d4f96ae06fb91a9db48865e963819